packages/commerce-sdk-react/src/hooks/ShopperCustomers/cache.ts

```diff
@@ -134,9 +135,37 @@ export const cacheUpdateMatrix: CacheUpdateMatrix<Client> = {
     },
     updateCustomerPassword: noop,
     updateCustomerProductList: TODO('updateCustomerProductList'),
-    updateCustomerProductListItem(customerId, {parameters}) {
+    updateCustomerProductListItem(customerId, {parameters}, response) {
+        const {itemId, listId} = parameters
         return {
-            update: [{queryKey: getCustomerProductListItem.queryKey(parameters)}],
+            update: [
+                {queryKey: getCustomerProductListItem.queryKey(parameters)},
+                // TODO: make separate PR that merges into develop
+                {
+                    // Also selectively update the lists
+                    queryKey: getCustomerProductLists.queryKey(parameters),
```

Comment:
Actually, after the list item is updated, the product lists become somewhat stale. For example, the `lastModified` time is no longer true.

So I prefer to also invalidate the product lists. Yes, most of the data can be redundant, but the refetch is going to be done in the background. In some way, I think of it as similar to stale-while-revalidate strategy.

Reply:
If we update and invalidate at the same time, does react query ever show the updated data? I thought that it just keeps the old data until the refetch completes. (I could very easily be wrong.)

Also, I just noticed that we update/invalidate the lists endpoint, but we just invalidate the list endpoint. We should follow the same pattern for both.

Reply:
You brought up good points there. To make things simpler, I've reverted the cache-update logic. 

This [article](https://tkdodo.eu/blog/mastering-mutations-in-react-query#direct-updates) mentions that it's safer and simpler to just invalidate, rather than update. And I can see that here in our case.

So what now? There's actually another way to achieve what I wanted: via local state update in the quantity picker component. Take a look at this commit for details: 487c77528ed0853bb77eb578edba849bbf102348